File: app/java/fas/src/main/java/com/example/accounting/domain/event/AccountUpdatedEvent.java
package com.example.accounting.domain.event;

import lombok.Getter;

import java.util.Map;

/**
 * 勘定科目更新イベント
 */
@Getter
public class AccountUpdatedEvent extends DomainEvent {
    private final String accountCode;
    private final Map<String, Object> oldValues;
    private final Map<String, Object> newValues;
    private final String userId;
    private final String userName;
    private final String ipAddress;

    public AccountUpdatedEvent(
            String accountCode,
            Map<String, Object> oldValues,
            Map<String, Object> newValues,
            String userId,
            String userName,
            String ipAddress) {
        super();
        this.accountCode = accountCode;
        this.oldValues = oldValues;
        this.newValues = newValues;
        this.userId = userId;
        this.userName = userName;
        this.ipAddress = ipAddress;
    }

    @Override
    public String getEventType() {
        return "AccountUpdated";
    }

    @Override
    public String getAggregateId() {
        return accountCode;
    }

    @Override
    public String getUserId() {
        return userId;
    }
}

File: app/csharp/AccountingSystem.Domain/Events/AccountUpdatedEvent.cs
namespace AccountingSystem.Domain.Events;

/// <summary>
/// 勘定科目更新イベント
/// </summary>
public record AccountUpdatedEvent : DomainEvent
{
    public required string AccountCode { get; init; }
    public required Dictionary<string, object> OldValues { get; init; }
    public required Dictionary<string, object> NewValues { get; init; }
    public required string UserId { get; init; }
    public required string UserName { get; init; }
    public string? IpAddress { get; init; }

    public override string GetEventType() => "AccountUpdated";
}
